src/test/java/io/xpring/ilp/DefaultIlpClientTest.java

```diff
@@ -0,0 +1,303 @@
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Unit tests for {@link io.xpring.ilp.DefaultIlpClient}
+ */
+public class DefaultIlpClientTest {
+
+  @Rule
+  public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  private GetBalanceResponse getBalanceResponse;
+  private CreateAccountResponse createAccountResponse;
+  private SendPaymentResponse sendPaymentResponse;
+  private GetAccountResponse getAccountResponse;
+
+  @Before
+  public void setUp() throws IOException {
+    getBalanceResponse = GetBalanceResponse.newBuilder()
+      .setAccountId("bob")
+      .setAssetCode("XRP")
+      .setAssetScale(9)
+      .setNetBalance(1000)
+      .setClearingBalance(10)
+      .setPrepaidAmount(100)
+      .build();
+
+    Map<String, String> customSettings = new HashMap<>();
+    customSettings.put("ilpOverHttp.incoming.auth_type", "SIMPLE");
+    customSettings.put("ilpOverHttp.incoming.auth_token", "gobbledygook");
+
+    createAccountResponse = CreateAccountResponse.newBuilder()
+      .setAccountRelationship("CHILD")
+      .setAssetCode("XRP")
+      .setAssetScale(9)
+      .putAllCustomSettings(customSettings)
+      .setAccountId("foo")
+      .setDescription("")
+      .setLinkType("ILP_OVER_HTTP")
+      .setIsConnectionInitiator(true)
+      .setIlpAddressSegment("foo")
+      .setBalanceSettings(CreateAccountResponse.BalanceSettings.newBuilder().build())
+      .setIsChildAccount(true)
+      .setIsInternal(false)
+      .setIsSendRoutes(true)
+      .setIsReceiveRoutes(false)
+      .setMaxPacketsPerSecond(0)
+      .setIsParentAccount(false)
+      .setIsPeerAccount(false)
+      .setIsPeerOrParentAccount(false)
+      .setPaymentPointer("$xpring.money.dev/foo")
+      .build();
+
+    getAccountResponse = GetAccountResponse.newBuilder()
+      .setAccountRelationship("CHILD")
+      .setAssetCode("XRP")
+      .setAssetScale(9)
+      .putAllCustomSettings(customSettings)
+      .setAccountId("foo")
+      .setDescription("")
+      .setLinkType("ILP_OVER_HTTP")
+      .setIsConnectionInitiator(true)
+      .setIlpAddressSegment("foo")
+      .setBalanceSettings(GetAccountResponse.BalanceSettings.newBuilder().build())
+      .setIsChildAccount(true)
+      .setIsInternal(false)
+      .setIsSendRoutes(true)
+      .setIsReceiveRoutes(false)
+      .setMaxPacketsPerSecond(0)
+      .setIsParentAccount(false)
+      .setIsPeerAccount(false)
+      .setIsPeerOrParentAccount(false)
+      .setPaymentPointer("$xpring.money.dev/foo")
+      .build();
+
+    sendPaymentResponse = SendPaymentResponse.newBuilder()
+      .setOriginalAmount(1000)
+      .setAmountDelivered(1000)
+      .setAmountSent(1000)
+      .setSuccessfulPayment(true)
+      .build();
+  }
+
+  @Test
+  public void minimalCreateIlpAccountTest() throws XpringException, IOException {
+    DefaultIlpClient client = getClient();
```

Comment:
Could you consider adding GIVEN, WHEN, THEN syntax here and in other tests? 

http://www.draconianoverlord.com/2017/11/28/given-when-then.html